### ISSUE TITLE
New array function to calculate PR-AUC

### ISSUE THREAD

**Issue by emmanuelsdias** [ClickHouse/ClickHouse#71788]
### Use case

I would like to have a way to compute the PR-AUC for my models, given their scores and labels. The function's output should match **sklearn**'s `average_precision_score`, just like **ClickHouse**'s `arrayAUC` matches **sklearn**'s `roc_auc_score`.

### Describe the solution you'd like

In a similar fashion to the `arrayAUC`, I would like to have a function which outputs the PR-AUC for a given set of scores and labels. 

I was thinking of implementing it as new function (`array_pr_auc`, `arrayPrecisionRecallAUC` or similar), since the current `arrayAUC` already outputs the specific ROC-AUC, not a generic AUC. 

I created this issue before attempting to implement it, to see if you folks agree with the creation of a new function to calculate it.

### Additional context

Ideally, AUC should compute the Area Under the Curve (AUC) for a given input of X and Y values. The current `arrayAUC`function calculates the AUC for the ROC curve, which plots the TPR (True Positive Rate) x FPR (False Positive Rate). So we could say that the `arrayAUC` actually computes the ROC-AUC, not just any AUC.

The ROC-AUC is a common way to evaluate the performance of binary classification models. It is preferred for balanced datasets (similar number of positive and negative labels). However, for unbalanced datasets, the PR-AUC is preferred. The PR-AUC calculates the AUC for the Precision Recall curve, which plots Precision (`TP / (TP + FP)`) x Recall (`TP / (TP + FN)`).

Here's an example of both curves mentioned above.

![image](https://github.com/user-attachments/assets/c450901b-e5f6-4b01-b3a7-9ca85dc3e65c)
[_(image source)_](https://juandelacalle.medium.com/how-and-why-i-switched-from-the-roc-curve-to-the-precision-recall-curve-to-analyze-my-imbalanced-6171da91c6b8)

For more info on this, check **sklearn**'s docs on it:
- [auc](https://scikit-learn.org/1.5/modules/generated/sklearn.metrics.auc.html) (AUC)
- [roc_auc_score](https://scikit-learn.org/1.5/modules/generated/sklearn.metrics.roc_auc_score.html) (ROC-AUC)
- [average_precision_score](https://scikit-learn.org/1.5/modules/generated/sklearn.metrics.average_precision_score.html) (PR-AUC)

**Comment by UnamedRus** [comment]
Discussion about another issue with different rules for AUC.
But, what is more important, IMO how naming should be changed.
https://github.com/ClickHouse/ClickHouse/discussions/65192#discussioncomment-9781601

IE, Rename current functions to arrayROCAUC (or whatever case), create alias to it with arrayAUC.
introduce new function arrayPRAUC
